### PR TITLE
SONARPY-670 FN on S2159 and S5655 due to missing simple third party s…

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
…ymbols from TypeShed

### CODE REVIEW

File: python-frontend/src/main/java/org/sonar/python/types/TypeShed.java [hunk 1]
```diff
@@ -120,39 +127,51 @@ private static void setDeclaredReturnType(Symbol symbol, FunctionDef functionDef
...
-      .filter(s -> s.fullyQualifiedName() != null)
+      .filter(s -> s.fullyQualifiedName() != null && s.fullyQualifiedName().startsWith(moduleName))
       .collect(Collectors.toMap(Symbol::fullyQualifiedName, Function.identity()));
   }
 
-  public static Set<Symbol> standardLibrarySymbols(String stdlibModuleName) {
-    if (!TypeShed.standardLibrarySymbols.containsKey(stdlibModuleName)) {
-      Map<String, Symbol> result = readTypeShedSymbols("typeshed/stdlib/2and3/" + stdlibModuleName + ".pyi", stdlibModuleName);
+  public static Set<Symbol> typeShedSymbols(String stdlibModuleName) {
+    if (!builtinsInitialized) {
```
Comment (andrea-guarino-sonarsource) [inline python-frontend/src/main/java/org/sonar/python/types/TypeShed.java:152]: Are we testing this? If I comment this code all unit tests are still passing

File: python-frontend/src/main/java/org/sonar/python/types/TypeShed.java [hunk 2]
```diff
@@ -120,39 +127,51 @@ private static void setDeclaredReturnType(Symbol symbol, FunctionDef functionDef
...
+      // Avoid cyclic dependencies when builtin and typing stubs import other stdlib symbols
+      return Collections.emptySet();
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply python-frontend/src/main/java/org/sonar/python/types/TypeShed.java]: For the record, the intent was to fix an infinite recursion we would face were we able to resolve the modules imported by `builtins.pyi`.

As we discussed, I will remove this and wait until we can reproduce the problem to fix it, most likely through a different implementation such as preventing the resolution of  imports when in special files (`builtins.pyi`; `typing.pyi` and `typing_extensions.pyi` namely).

File: python-frontend/src/test/java/org/sonar/python/types/TypeShedTest.java [hunk 3]
```diff
@@ -78,19 +78,28 @@ public void typing_module() {
...
 
-    Map<String, Symbol> imaplibSymbols = TypeShed.standardLibrarySymbols("imaplib").stream().collect(Collectors.toMap(Symbol::name, Function.identity()));
+    Map<String, Symbol> imaplibSymbols = TypeShed.typeShedSymbols("imaplib").stream().collect(Collectors.toMap(Symbol::name, Function.identity()));
     assertThat(imaplibSymbols).isNotEmpty();
     assertThat(imaplibSymbols.values()).allMatch(symbol -> symbol.usages().isEmpty());
   }
+
+  @Test
+  public void third_party_symbols() {
+    Map<String, Symbol> emojiSymbols = TypeShed.typeShedSymbols("emoji").stream().collect(Collectors.toMap(Symbol::name, Function.identity()));
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply python-frontend/src/test/java/org/sonar/python/types/TypeShedTest.java]: Fixed in 34649bbcc8c3ea0b46c5cecd74b999085e3d6542

File: python-frontend/src/main/java/org/sonar/python/types/TypeShed.java [hunk 4]
```diff
@@ -120,39 +125,47 @@ private static void setDeclaredReturnType(Symbol symbol, FunctionDef functionDef
...
       })
-      .filter(s -> s.fullyQualifiedName() != null)
+      .filter(s -> s.fullyQualifiedName() != null && s.fullyQualifiedName().startsWith(moduleName))
       .collect(Collectors.toMap(Symbol::fullyQualifiedName, Function.identity()));
   }
 
-  public static Set<Symbol> standardLibrarySymbols(String stdlibModuleName) {
-    if (!TypeShed.standardLibrarySymbols.containsKey(stdlibModuleName)) {
-      Map<String, Symbol> result = readTypeShedSymbols("typeshed/stdlib/2and3/" + stdlibModuleName + ".pyi", stdlibModuleName);
+  public static Set<Symbol> typeShedSymbolsForModule(String stdlibModuleName) {
```
Comment (andrea-guarino-sonarsource) [inline python-frontend/src/main/java/org/sonar/python/types/TypeShed.java:149]: Minor comment: I think we don't need to prefix the method name with "typeshed" as it's part of TypeShed class.

Comment (guillaume-dequenne-sonarsource) [reply python-frontend/src/main/java/org/sonar/python/types/TypeShed.java:149]: If that's okay for you, I'll keep the name here, as the next PR I'm preparing includes further refactoring of this method, I'll take the opportunity to review the names consistency at the same time.

**Pull request merged**